LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,496 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Security.Cryptography.X509Certificates;
+using System.IO;
+using Docker.DotNet;
+using Docker.DotNet.Models;
+using Microsoft.Azure.Gaming.LocalMultiplayerAgent.DeploymentTool;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Azure.Gaming.AgentInterfaces;
+using Microsoft.Azure.Gaming.LocalMultiplayerAgent.Config;
+using Newtonsoft.Json;
+using PlayFab;
+using PlayFab.MultiplayerModels;
+
+namespace Microsoft.Azure.Gaming.LocalMultiplayerAgent.MPSDeploymentTool
+{
+    public class DeploymentScript
+    {
+        private readonly MultiplayerSettings settings;
+        private readonly DeploymentSettings deploymentSettings;
+
+        public DeploymentScript(MultiplayerSettings multiplayerSettings)
+        {
+            settings = multiplayerSettings ?? throw new ArgumentNullException(nameof(multiplayerSettings));
+            deploymentSettings = JsonConvert.DeserializeObject<DeploymentSettings>(File.ReadAllText("DeploymentTool/DeploymentSettings.json"));
+        }
+
+        public async Task RunScriptAsync()
+        {
+            Console.WriteLine("Deploying to PlayFab Multiplayer Servers...\n");
+
+            var auth = await PlayFabAuthentication();
+
+            ErrorCheck(auth.Error);
+
+            DeploymentSettingsValidator validator = new DeploymentSettingsValidator(deploymentSettings);
```

Comment:
validating the deployment settings is faster than authenticating, I'd do that first :)